openmetadata-ui/src/main/resources/ui/src/components/Modals/RelatedTermsModal/RelatedTermsModal.tsx

```diff
@@ -59,11 +61,11 @@ const RelatedTermsModal = ({
   const suggestionSearch = (searchText = '') => {
     setIsLoading(true);
     searchData(searchText, 1, PAGE_SIZE, '', '', '', SearchIndex.GLOSSARY)
-      .then((res: SearchResponse) => {
+      .then((res) => {
         const termResult = (
           formatSearchGlossaryTermResponse(
-            res?.data?.hits?.hits || []
-          ) as FormattedGlossaryTermData[]
+            (res?.data as SearchResponse<SearchIndex.GLOSSARY>)?.hits?.hits
+          ) as GlossaryTerm[]
```

Comment:
Do we need this as we are using the `searchData` method?
Is searchData not returning the result with the proper type?

Reply:
Done removed type casting